latch/types/metadata.py

```diff
@@ -335,14 +335,23 @@ def dict(self):
 
 
 @dataclass
-class SnakemakeFileParameter(LatchParameter):
-    type: Optional[Union[Type[LatchFile], Type[LatchDir]]] = None
+class SnakemakeParameter(LatchParameter):
+    type: Optional[
+        Union[
+            Type[LatchFile],
+            Type[LatchDir],
+            Type[str],
+            Type[Enum],
+        ]
+    ] = None
     """
     The python type of the parameter.
     """
     path: Optional[Path] = None
     """
     The path where the file passed to this parameter will be copied.
+
+    Only relevant if self.type is `LatchFile` or `LatchDir`.
```

Comment:
kinda awful API design but eh

maybe have `SnakemakeFileParameter` be a subclass and remove `path` from normal `SnakemakeParameter`?